### PR TITLE
 add "Open Rule link in browser" 

### PR DESCRIPTION
**Pull request by qxo**
This PR will index the checks.html content, link rule to the detail url.
ie:
```
AbstractClassName -->  https://checkstyle.sourceforge.io/config_naming.html#AbstractClassName
```

we can change the rule info url：
```
-Declipsecs_rules_url=https://checkstyle.sourceforge.io/checks.html
```

### CODE REVIEW

File: net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/stats/views/RuleLinkHelper.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import net.sf.eclipsecs.core.util.CheckstyleLog;
+
+/**
+ * Checkstyle rule link helper.
+ */
+public class RuleLinkHelper {
+
+  private static final String RULES_URL = System.getProperty("eclipsecs_rules_url",
+          "https://checkstyle.sourceforge.io/checks.html");
```
Comment (romani) [inline net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/stats/views/RuleLinkHelper.java:45]: should be changed to `checkstyle.org` it is primary domain.

**Review by romani — CHANGES_REQUESTED** [review]
please fix CI.

item:

**Comment by romani** [comment]
CI failures https://travis-ci.org/github/checkstyle/eclipse-cs/jobs/727087072#L962 please fix

**Comment by romani** [comment]
@qxo ,
you will not need to cache anything ones https://github.com/checkstyle/eclipse-cs/pull/257 is merged.
Please help to test this PR, it will simplify a lot your PR and any other problems.

**Comment by qxo** [comment]
> @qxo ,
> you will not need to cache anything ones #257 is merged.
> Please help to test this PR, it will simplify a lot your PR and any other problems.

ok, if I had the time.

What's the #257 problem which we should focus on?

**Comment by romani** [comment]
> What's the #257 problem which we should focus on?

we try to get all data about Checks from checkstyle library itself as checkstyle jar now contain almost all metadata(all docs) that is required for plugin and users.

**Comment by Bananeweizen** [comment]
I'd like to close this in favour of #568. I'd avoid trying to index any online URL to get the available help URLs.
We could eventually resurrect the "Open URL" actions from this PR.

**Pull request closed without merging**